### PR TITLE
[LC-972] Multi Node from Genesis

### PR DESCRIPTION
**Pull request by metalg0su**
# Goal
- Runnable loopchain 3.0 between multiple nodes (from genesis only)

### CODE REVIEW

File: loopchain/blockchain/blocks/v1_0/block_factory.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def _process_added_transactions(self, block_builder, added_transactions):
     def create_none_data(self, epoch_num: int, round_num: int, proposer_id: bytes) -> Block:
         return self._create_unreal_data(epoch_num, round_num, proposer_id, _hash=Block.NoneData)
 
+
```
Comment (metalg0su) [inline loopchain/blockchain/blocks/v1_0/block_factory.py:151]: Remove

File: loopchain/blockchain/blockchain.py [hunk 2]
```diff
@@ -1072,10 +1072,19 @@ def block_loads(self, block_dumped: bytes) -> 'Block':
...
-        block_serializer = BlockSerializer.new(block_version, self.__tx_versioner)
-        return block_serializer.deserialize(block_serialized)
+        try:
+            block_height = self.__block_versioner.get_height(block_serialized)
+            block_version = self.__block_versioner.get_version(block_height)
+            block_serializer = BlockSerializer.new(block_version, self.__tx_versioner)
+            block = block_serializer.deserialize(block_serialized)
+        except AttributeError:  # no serializer above v1.0
+            from loopchain.blockchain.blocks.v1_0 import Block as Block_V1_0
+            block = Block_V1_0._deserialize(**block_serialized)
```
Comment (hyerim-kim) [inline loopchain/blockchain/blockchain.py:1082]: Is there any reason to deserialize block directly, without using `BlockSerializer` ?

File: loopchain/consensus/runner.py [hunk 3]
```diff
@@ -283,6 +284,21 @@ def update_status(self, peer: str, height: int):
...
+        if block_hash == Hash32.empty():  # On Genesis Block
+            validators = blockchain.find_preps_by_roothash(ChannelProperty().crep_root_hash)
+        else:
+            block: Block = blockchain.find_block_by_hash32(block_hash)
+            validators_hash = block.header.next_validators_hash
+            validators = blockchain.find_preps_by_roothash(validators_hash)
+
+        return self._convert_to_external_addr(validators)
+
+    def _convert_to_external_addr(self, validators: List[dict]) -> List[ExternalAddress]:
```
Comment (hyerim-kim) [inline loopchain/consensus/runner.py:299]: just suggestion. 😅 
```suggestion
    def _convert_to_external_address(self, validators: List[dict]) -> List[ExternalAddress]:
```

File: loopchain/blockchain/blockchain.py (hunk 2 above)
Comment (windies21) [reply loopchain/blockchain/blockchain.py:1082]: I have a same question too.

Comment (metalg0su) [reply loopchain/blockchain/blockchain.py:1082]: I have a same question too. 😅

Comment (metalg0su) [reply loopchain/blockchain/blockchain.py]: Fixed!

**Pull request merged**